### DISCUSSION TITLE
Understanding of the compress option

### DISCUSSION THREAD

**Discussion by nilsm0hr** [aws-powertools/powertools-lambda-python#1885] [General]
Hi, 

I tired to understand the compress option in PowerTools. We have started to implement powertools on our functions. Some of them compressed the Lambda response in our python code and returned the base64 payload to enable larger payloads.

The documentation suggests that you just return an object, rather than the Response object powertools provides. Looking at the source code, I managed to use the powertools response object after I dumped my json response to a string. 

Would it be possible to make the docs a bit clearer around this? Not quite sure if I am misunderstanding something here.

https://awslabs.github.io/aws-lambda-powertools-python/2.7.1/core/event_handler/api_gateway/#compress

Thanks

**Comment by rubenfonseca** [comment]
Hi @nilsm0hr thank you so much for opening this up! I agree that we need to improve the documentation on how to use compressed objects with the `Response` object. Would you mind opening a [new documentation improvement issue](https://github.com/awslabs/aws-lambda-powertools-python/issues/new?created_from_discussion_number=1885) for us to track and prioritize that?

Thank you!

**Reply by heitorlessa** [reply]
We can covert this discussion into an issue too - don't see the button on mobile but surely is on the web UI.

We can do tomorrow regardless

**Reply by rubenfonseca** [reply]
Done!

**Comment by rubenfonseca** [comment]
Converted the question to an issue https://github.com/awslabs/aws-lambda-powertools-python/issues/1890

**Comment by nilsm0hr** [comment]
Thanks guys! really appreciate it

**Comment by heitorlessa** [comment]
Closing as it was released back then: https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v2.17.0
